dra-evolution/pkg/api/claim_types.go

```diff
@@ -1,379 +1,670 @@
 package api
 
 import (
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 // DeviceClass is a vendor or admin-provided resource that contains
 // contraint and configuration information. Essentially, it is a re-usable
 // collection of predefined data that device claims may use.
 // Cluster scoped.
-type DeviceClass struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+type ResourceClass struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// ControllerName defines the name of the dynamic resource driver that is
+	// used for allocation of a ResourceClaim that uses this class. If empty,
+	// structured parameters are used for allocating claims using this class.
+	//
+	// Resource drivers have a unique name in forward domain order
+	// (acme.example.com).
+	//
+	// This is an alpha field and requires enabling the DRAControlPlaneController
+	// feature gate.
+	//
+	// +optional
+	ControllerName string `json:"controllername,omitempty"`
+
+	// Only nodes matching the selector will be considered by the scheduler
+	// when trying to find a Node that fits a Pod when that Pod uses
+	// a ResourceClaim that has not been allocated yet.
+	//
+	// Setting this field is optional. If null, all nodes are candidates.
+	// +optional
+	SuitableNodes *v1.NodeSelector `json:"suitableNodes,omitempty" protobuf:"bytes,4,opt,name=suitableNodes"`
+
+	// Claim contains options that apply to the entire claim.
+	Claim ResourceClassClaimOptions `json:"claim,omitempty"`
+
+	// Request contains options that apply to requests in a claim.
+	Request ResourceClassRequestOptions `json:"request,omitempty"`
```

Comment:
I don't think requests belong in classes.

Reply:
See https://github.com/kubernetes-sigs/wg-device-management/pull/14/files#r1611132210